client/src/components/Suggestions/UpdateMonumentSuggestions/UpdateMonumentSuggestion/UpdateMonumentSuggestion.js

```diff
@@ -0,0 +1,77 @@
+import * as React from 'react';
+import './UpdateMonumentSuggestion.scss';
+import { Card } from 'react-bootstrap';
+import MonumentUpdate from '../../../Monument/Update/MonumentUpdate';
+import { Link } from 'react-router-dom';
+
+export default class UpdateMonumentSuggestion extends React.Component {
+
+    determineDateTypeForUpdate() {
+        const { suggestion } = this.props;
+
+        if (suggestion.newDate) {
+            return 'exact-date';
+        }
+
+        if (suggestion.newMonth) {
+            return 'month-year';
+        }
+
+        return 'year';
+    }
+
+    buildUpdate() {
+        const { suggestion } = this.props;
+
+        return {
+            ...suggestion,
+            date: {
+                type: this.determineDateTypeForUpdate(),
+                newYear: suggestion.newYear,
+                newMonth: suggestion.newMonth,
+                newDate: suggestion.newDate
+            },
+            newLatitude: suggestion.newLatitude ? suggestion.newLatitude.toString() : '',
+            newLongitude: suggestion.newLongitude ? suggestion.newLongitude.toString() : '',
+            newMaterials: suggestion.newMaterialsJson ? JSON.parse(suggestion.newMaterialsJson) : undefined,
+            newTags : suggestion.newTagsJson ? JSON.parse(suggestion.newTagsJson) : undefined,
+            deletedReferenceIds: suggestion.deletedReferenceIdsJson ? JSON.parse(suggestion.deletedReferenceIdsJson) : undefined,
+            updatedReferenceUrlsById: suggestion.updatedReferenceUrlsByIdJson ? JSON.parse(suggestion.updatedReferenceUrlsByIdJson) : undefined,
+            newReferenceUrls: suggestion.newReferenceUrlsJson ? JSON.parse(suggestion.newReferenceUrlsJson) : undefined,
+            addedImages: suggestion.newImageUrlsJson ? JSON.parse(suggestion.newImageUrlsJson).map(url => ({url})) : undefined,
+            deletedImageUrls: suggestion.deletedImageUrlsJson ? JSON.parse(suggestion.deletedImageUrlsJson) : undefined,
```

Comment:
```suggestion
            newLatitude: (suggestion.newLatitude || '').toString(),
            newLongitude: (suggestion.newLongitude || '').toString(),
            newMaterials: suggestion.newMaterialsJson && JSON.parse(suggestion.newMaterialsJson),
            newTags : suggestion.newTagsJson && JSON.parse(suggestion.newTagsJson),
            deletedReferenceIds: suggestion.deletedReferenceIdsJson && JSON.parse(suggestion.deletedReferenceIdsJson),
            updatedReferenceUrlsById: suggestion.updatedReferenceUrlsByIdJson && JSON.parse(suggestion.updatedReferenceUrlsByIdJson),
            newReferenceUrls: suggestion.newReferenceUrlsJson && JSON.parse(suggestion.newReferenceUrlsJson),
            addedImages: suggestion.newImageUrlsJson && JSON.parse(suggestion.newImageUrlsJson).map(url => ({url})),
            deletedImageUrls: suggestion.deletedImageUrlsJson && JSON.parse(suggestion.deletedImageUrlsJson),
```

Reply:
If you use `&&` in this context, JS will return the first value if it's falsey or the second value if it's true, rather than returning true/false like Java would
```javascript
let a = undefined;
console.log(a && JSON.parse(a));
>> undefined
a = '{}';
console.log(a && JSON.parse(a));
>> {}
```